core/io/resource_loader.cpp

```diff
@@ -1053,8 +1053,10 @@ void ResourceLoader::clear_thread_load_tasks() {
 		thread_load_mutex.lock();
 	}
 
-	for (KeyValue<String, LoadToken *> &E : user_load_tokens) {
-		memdelete(E.value);
+	for (HashMap<String, LoadToken *>::Iterator I = user_load_tokens.begin(); I;) {
+		HashMap<String, LoadToken *>::Iterator current = I;
+		++I;
+		memdelete(current->value);
 	}
```

Comment:
The fix is very good. That said, I'd suggest this other style, more in line with other similar cases in Godot.

```suggestion
	while (user_load_tokens.begin()) {
		// User load tokens remove themselves from the map on destruction.
		memdelete(*user_load_tokens.begin());
	}
```

Reply:
As long as the order of destruction doesn't matter I'd say to go with this indeed